### PR TITLE
reintroduce window in days, log warning when sampling occurs

### LINKED ISSUE #8570: Source Google Analytics: Improve sampling awareness
## Problem
The `window_in_days` parameters was removed from source Google Analytics specification to simplify configuration, it now defaults to 90 days. This decision comes from this issue https://github.com/airbytehq/airbyte/issues/7850 and was implemented in this PR https://github.com/airbytehq/airbyte/pull/8175.

This parameters could be used as a workaround to avoid Google Analytics API sampling that can lead to return inconsistent results according to user expectation. Google Analytics samples results when reports has > 500000 records. Downsizing reports with a custom `window_in_days` parameters allow the avoidance sampling while increasing the sync duration. 

Moreover it's not currently explicit for users that a report they ingest is sampled or not. This information is returned by Google Analytics API so we should be able to display warning logs / automatically reduce the report size to solve this sampling problem.

## Solution
I'd suggest the following:
* Display warning logs that a report was sampled during read operation 
* Re-add `windows_in_day` parameters (or rename in to `batch_size`) + document its impact in the specification description field.
* Or make the connector smart enough to adapt the batch size if the consumed reports appears to be sampled.

## Reference
I opened this issue following a Slack [thread](https://airbytehq.slack.com/archives/C01MFR03D5W/p1638643722342600?thread_ts=1637860057.477300&cid=C01MFR03D5W) with @folusoogunlana

### PR DESCRIPTION
**Pull request by eliziario**
# Improve sampling awareness and reintroduce parameter for mitigation in source google analytics

Fixes #8570

## Reason
Google analytics may return sampled data when there are too many sessions (about 500k) in a given period. 
Also, it may return provisional data (isDataGolden=false) meaning that the report can change if requested again in the future. 
The Sampling issue can be partially mitigated (but not completely eliminated) by using smaller values for window_in_days parameter. 
For those cases when even using window_in_days equal to one is not enough to avoid sampling (when there are more than 500k sessions or other dimension values in a report in a day) the user should be warned in the logs. 
Also, when isDataGolden=false the user should be warned in logs

## Confirmation
 - Generating millions of data records in an analytics api using the instrumentation api in a script reproduces the issue, which is also described in the documentation

## How does the code change in the PR fix the issue?
 - Changed the default value for the window_in_days parameter
- Re-exposed the window_in_days parameter on configuration for better user ergonomics based on their usage profile (users with lots of analytics data will choose smaller values to avoid sampling, users with less data will prefer higher values to improve initial sync speed. 
-  Added logging warning for when isDataGolden is false
-  Added logging warning for when analytics api is sampling data.  
---

### Recommended reading order
1. source.py
2. spec.json
3. unit_test.py

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py [hunk 1]
```diff
@@ -418,7 +431,7 @@ class GoogleAnalyticsServiceOauth2Authenticator(Oauth2Authenticator):
     def __init__(self, config):
         self.credentials_json = json.loads(config["credentials_json"])
         self.client_email = self.credentials_json["client_email"]
-        self.scope = "https://www.googleapis.com/auth/analytics.readonly"
+        self.scope = "https://www.googleapis .com/auth/analytics.readonly"
```
Comment (sergei-solonitcyn) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py:434]: misprint
```suggestion
        self.scope = "https://www.googleapis.com/auth/analytics.readonly"
```

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py [hunk 2]
```diff
@@ -91,7 +98,7 @@ class GoogleAnalyticsV4Stream(HttpStream, ABC):
     def __init__(self, config: Dict):
         super().__init__(authenticator=config["authenticator"])
         self.start_date = config["start_date"]
-        self.window_in_days = config.get("window_in_days", 90)
+        self.window_in_days = config.get("window_in_days", 1)
```
Comment (sergei-solonitcyn) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py:101]: Maybe it's good idea to mention this behavior in the config descriotion.

**Review by sergei-solonitcyn — CHANGES_REQUESTED** [review]

**Comment by sergei-solonitcyn** [comment]
Please, don't forget to run tests and checks before every comit and especially before pushing.

**Review by vitaliizazmic — COMMENTED** [review]
LGTM, only requested small changes is needed.

**Comment by sergei-solonitcyn** [comment]
Made some updates, but tests are still broken and there are suspections that the code as well.
Will reassign it and finish the work.

**Review by augan-rymkhan — APPROVED** [review]

**Review by sergei-solonitcyn — APPROVED** [review]

**Pull request merged**